crates/recursor/src/error.rs

```diff
@@ -166,16 +166,10 @@ impl From<Error> for String {
 
 impl From<ResolveError> for Error {
     fn from(e: ResolveError) -> Self {
-        if let Some(ProtoErrorKind::NoRecordsFound { soa, ns, .. }) =
+        if let Some(ProtoErrorKind::NoRecordsFound { ns: Some(ns), .. }) =
             e.proto().map(ProtoError::kind)
         {
-            if let Some(ns) = ns {
-                ErrorKind::ForwardNS(ns.clone()).into()
-            } else if let Some(soa) = soa {
-                ErrorKind::Forward(soa.name().clone()).into()
-            } else {
-                ErrorKind::Resolve(e).into()
-            }
+            ErrorKind::ForwardNS(ns.clone()).into()
```

Comment:
I think it (or something like it) is needed to copy the ns record data from the ResolveError

Reply:
I don't know why? It looks like we have ownership of the `ResolveError` here?

Reply:
agree, this clone is probably unnecessary... I'm surprised clippy isn't warning about that?

Reply:
Removing the clone and dereferencing:

>                 ErrorKind::ForwardNS(*ns).into()

yields:

```
error[E0507]: cannot move out of `*ns` which is behind a shared reference
   --> crates/recursor/src/error.rs:173:38
    |
173 |                 ErrorKind::ForwardNS(*ns).into()
    |                                      ^^^ move occurs because `*ns` has type `Arc<[ForwardNSData]>`, which does not implement the `Copy`
```

Reply:
Right, that's because this is leaning on `ResolveError::proto()` yield an `Option<&ProtoError>` out of the `ResolveError` -- we should have a way to take the `ResolveErrorKind` out of the `ResolveError`, maybe an `impl From<ResolveError> for ResolveErrorKind`?